lib/customer.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'takeaway'
+
+class Customer
+
+  attr_accessor :order_basket, :order_total, :total_items, :takeaway
+
+  def initialize(klass = Takeaway)
+    @order_basket = {}
+    @order_total = 0
+    @total_items = 0
+    @takeaway = klass.new
+  end
+
+  def view_menu
+    @takeaway.menu.each { |key, value| p "#{key} - #{value}"}
+  end
+
+  def select_item(item, quantity)
+    fail 'Sorry, that item is not on the menu.' unless @takeaway.menu.has_key?(item)
```

Comment:
Metrics/LineLength: Line is too long. [84/80]<br>Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`.